catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessor.java

```diff
@@ -0,0 +1,219 @@
+   */
+  private static DMSCoordinates parseDms(final String dmsString) {
+    final Matcher matcher = PATTERN_DMS_COORDINATES.matcher(dmsString);
+
+    if (matcher.matches()) {
+      final DMSComponent lat =
+          new DMSComponent(
+              matcher.group(DEGREES_LAT_GROUP),
+              matcher.group(MINUTES_LAT_GROUP),
+              matcher.group(SECONDS_LAT_GROUP),
+              matcher.group(DIRECTION_LAT_GROUP),
+              new DecimalFormat("00"));
+      final DMSComponent lon =
+          new DMSComponent(
+              matcher.group(DEGREES_LON_GROUP),
+              matcher.group(MINUTES_LON_GROUP),
+              matcher.group(SECONDS_LON_GROUP),
+              matcher.group(DIRECTION_LON_GROUP),
+              new DecimalFormat("000"));
+      return new DMSCoordinates(lat, lon);
+    }
+    return null;
+  }
+
+  private static String normalizedDmsString(final String dmsString) {
+
+    final DMSCoordinates dmsCoordinates = parseDms(dmsString);
+    final List<DMSComponent> dmsParts = Arrays.asList(dmsCoordinates.lat, dmsCoordinates.lon);
+    final NumberFormat minutesSecondsFormat = new DecimalFormat("00.###");
+
+    final StringBuilder dmsBuilder = new StringBuilder();
+    for (DMSComponent dmsPart : dmsParts) {
+      dmsBuilder
+          .append(dmsPart.degreeFormat.format(dmsPart.degrees))
+          .append("\u00B0")
+          .append(minutesSecondsFormat.format(dmsPart.minutes))
+          .append("\'")
+          .append(minutesSecondsFormat.format(dmsPart.seconds))
+          .append("\"")
+          .append(dmsPart.direction.toUpperCase())
+          .append(" ");
+    }
+    return dmsBuilder.toString().trim();
+  }
+
+  private static LatLon dmsToLatLon(final String dmsString) {
+    final DMSCoordinates dmsCoordinates = parseDms(dmsString);
+    if (dmsCoordinates == null) {
+      return null;
+    }
+
+    final DMSComponent dmsLat = dmsCoordinates.lat;
+    final DMSComponent dmsLon = dmsCoordinates.lon;
+    final int latModifier = dmsLat.direction.equalsIgnoreCase("N") ? 1 : -1;
+    final int lonModifier = dmsLon.direction.equalsIgnoreCase("E") ? 1 : -1;
+    final Double lat = toDecimalDegrees(dmsLat) * latModifier;
+    final Double lon = toDecimalDegrees(dmsLon) * lonModifier;
+
+    return (LatLon.isValidLatitude(lat) && LatLon.isValidLongitude(lon))
+        ? new LatLon(lat, lon)
+        : null;
+  }
+
+  private static Double toDecimalDegrees(final DMSComponent dms) {
+    return dms.degrees + dms.minutes / 60.0 + dms.seconds / 3600.0;
+  }
+
+  @Nullable
+  private LiteralSuggestion getDmsSuggestion(final String query) {
+    final List<LatLon> dmsCoordinates = new ArrayList<>();
+    final Matcher matcher = PATTERN_DMS_COORDINATES.matcher(query);
+    final StringBuilder nameBuilder = new StringBuilder("DMS:");
+
+    int start = 0;
+    while (matcher.find()) {
+      final String group = query.substring(start, matcher.end());
+      final LatLon parsedDms = dmsToLatLon(group);
```

Comment:
❓ Instead of `dmsToLatLon` which is a pretty big jump, could we create a static method on `DMSCoordinate` called `createFromString` instead? Then, if `group` was renamed to `dmsString`, it would look like this:
```
final DMSCoordinate dms = DMSCoordinate.createFromString(dmsString);
final LatLon latlon = LatLon.createIfValid(dms.lat.toDecimalDegree(), dms.lon.toDecimalDegree());
```
We would have to add the `@Nullable createIfValid` method to the `LatLon` class. We also should update the `LatLon` constructor to check our inputs so we always build a valid `LatLon`.